great_expectations/core/batch.py

```diff
@@ -412,6 +413,30 @@ def __init__(
         self._runtime_parameters = runtime_parameters
         self._batch_identifiers = batch_identifiers
 
+    def __deepcopy__(self, memo):
```

Comment:
❤️

Reply:
Actually, can we add a condition here to NOT deep copy if the runtime parameters are specifying a spark dataframe?  Spark dataframes are immutable, and there is no harm in skipping the deep copy.  Conversely, such a deep copy throws errors when running Checkpoints in Databricks.